sonar-core/src/main/java/org/sonar/core/platform/PluginInfo.java

```diff
@@ -0,0 +1,392 @@
+    return useChildFirstClassLoader;
+  }
+
+  @CheckForNull
+  public String getBasePlugin() {
+    return basePlugin;
+  }
+
+  public boolean isCore() {
+    return core;
+  }
+
+  @CheckForNull
+  public String getImplementationBuild() {
+    return implementationBuild;
+  }
+
+  public Set<RequiredPlugin> getRequiredPlugins() {
+    return requiredPlugins;
+  }
+
+  public PluginInfo setName(@Nullable String name) {
+    this.name = Objects.firstNonNull(name, this.key);
+    return this;
+  }
+
+  public PluginInfo setVersion(Version version) {
+    this.version = version;
+    return this;
+  }
+
+  public PluginInfo setMinimalSqVersion(@Nullable Version v) {
+    this.minimalSqVersion = v;
+    return this;
+  }
+
+  /**
+   * Required
+   */
+  public PluginInfo setMainClass(String mainClass) {
+    this.mainClass = mainClass;
+    return this;
+  }
+
+  public PluginInfo setDescription(@Nullable String description) {
+    this.description = description;
+    return this;
+  }
+
+  public PluginInfo setOrganizationName(@Nullable String s) {
+    this.organizationName = s;
+    return this;
+  }
+
+  public PluginInfo setOrganizationUrl(@Nullable String s) {
+    this.organizationUrl = s;
+    return this;
+  }
+
+  public PluginInfo setLicense(@Nullable String license) {
+    this.license = license;
+    return this;
+  }
+
+  public PluginInfo setHomepageUrl(@Nullable String s) {
+    this.homepageUrl = s;
+    return this;
+  }
+
+  public PluginInfo setIssueTrackerUrl(@Nullable String s) {
+    this.issueTrackerUrl = s;
+    return this;
+  }
+
+  public PluginInfo setUseChildFirstClassLoader(boolean b) {
+    this.useChildFirstClassLoader = b;
+    return this;
+  }
+
+  public PluginInfo setBasePlugin(@Nullable String s) {
+    this.basePlugin = s;
+    return this;
+  }
+
+  public PluginInfo setCore(boolean b) {
+    this.core = b;
+    return this;
+  }
+
+  public PluginInfo setImplementationBuild(@Nullable String implementationBuild) {
+    this.implementationBuild = implementationBuild;
+    return this;
+  }
+
+  public PluginInfo addRequiredPlugin(RequiredPlugin p) {
+    this.requiredPlugins.add(p);
+    return this;
+  }
+
+  /**
+   * Find out if this plugin is compatible with a given version of SonarQube.
+   * The version of SQ must be greater than or equal to the minimal version
+   * needed by the plugin.
+   */
+  public boolean isCompatibleWith(String sqVersion) {
+    if (null == this.minimalSqVersion) {
+      // no constraint defined on the plugin
+      return true;
+    }
+
+    Version effectiveMin = Version.create(minimalSqVersion.getName()).removeQualifier();
+    Version actualVersion = Version.create(sqVersion).removeQualifier();
+    return actualVersion.compareTo(effectiveMin) >= 0;
+  }
+
+  @Override
+  public String toString() {
+    return String.format("[%s]", SLASH_JOINER.join(key, version, implementationBuild));
+  }
+
+  @Override
+  public int compareTo(PluginInfo other) {
```

Comment:
:new: ![CRITICAL](https://raw.githubusercontent.com/henryju/image-hosting/master/critical.png)Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1210)